enterprise/backend/src/metabase_enterprise/sso/integrations/sso_utils.clj

```diff
@@ -29,11 +30,36 @@
    [:sso_source       [:enum :saml :jwt]]
    [:login_attributes [:maybe :map]]])
 
+(defn- maybe-throw-user-provisioning
+  [user-provisioning-type]
+  (when (not user-provisioning-type)
+    (throw (ex-info (trs "Sorry, but you''ll need a {0} account to view this page. Please contact your administrator."
+                         (u/slugify (public-settings/site-name))) {}))))
+
+(defmulti check-user-provisioning
+  "If `user-provisioning-enabled?` is false, then we should throw an error when attempting to create a new user."
+  {:arglists '([model])}
+  keyword)
+
+(defmethod check-user-provisioning :saml
+  [_]
+  (maybe-throw-user-provisioning (sso-settings/saml-user-provisioning-enabled?)))
+
+(defmethod check-user-provisioning :ldap
+  [_]
+  (maybe-throw-user-provisioning (sso-settings/ldap-user-provisioning-enabled?)))
+
+(defmethod check-user-provisioning :jwt
+  [_]
+  (maybe-throw-user-provisioning (sso-settings/jwt-user-provisioning-enabled?)))
+
+
 (mu/defn create-new-sso-user!
   "This function is basically the same thing as the `create-new-google-auth-user` from `metabase.models.user`. We need
   to refactor the `core_user` table structure and the function used to populate it so that the enterprise product can
-  reuse it"
+  reuse it."
   [user :- UserAttributes]
+  (check-user-provisioning (:sso_source user))
```

Comment:
nit - I think it might be a little clearer to just move this check up a level, and only call `create-new-sso-user!` if provisioning is enabled for the given SSO type.

It keeps the behavior of `create-new-sso-user!` consistent with its name, and makes the control flow a bit easier to follow when you're reading the implementations of `fetch-or-create-user!`.

Won't block the PR on this since the logic should work fine either way, but just my 2¢

Reply:
Yea that makes a ton of sense